src/auth/src/credentials/service_account_credential.rs

```diff
@@ -54,18 +66,16 @@ impl std::fmt::Debug for ServiceAccountInfo {
     }
 }
 
-#[allow(dead_code)] // TODO(#679) - implementation in progress
 #[derive(Debug)]
-pub(crate) struct ServiceAccountCredential<T>
```

Comment:
why should we make these structs visible outside?

Reply:
Ditto, I made it so `credentials.rs` cannot use this type directly.